pageserver/src/page_service.rs

```diff
@@ -556,25 +555,10 @@ impl PageServerHandler {
     {
         debug_assert_current_span_has_tenant_and_timeline_id_no_shard_id();
 
-        let tenant = self
-            .get_active_tenant_with_timeout(tenant_id, ShardSelector::First, ACTIVE_TENANT_TIMEOUT)
-            .await?;
-
-        // Make request tracer if needed
-        let mut tracer = if tenant.get_trace_read_requests() {
-            let connection_id = ConnectionId::generate();
-            let path =
-                tenant
-                    .conf
-                    .trace_path(&tenant.tenant_shard_id(), &timeline_id, &connection_id);
-            Some(Tracer::new(path))
-        } else {
-            None
-        };
-
         // switch client to COPYBOTH
         pgb.write_message_noflush(&BeMessage::CopyBothResponse)?;
-        self.flush_cancellable(pgb, &tenant.cancel).await?;
```

Comment:
This is the core change of this PR.

We can't merge the switch to `NO_CANCEL` at this point because it might exacerbate the problem that causes stuck detaches: https://github.com/neondatabase/neon/issues/7427#issuecomment-2220523013

Reply:
https://github.com/neondatabase/neon/pull/8338/commits/8654c2d6dea6e190e2dc4325cee24f43ccaed7b0